projects/plugins/boost/app/lib/critical-css/Critical_CSS_State.php

```diff
@@ -109,7 +109,13 @@ private function get_state_transient() {
 
 	public function maybe_set_status() {
 		if ( $this->get_total_providers_count() === $this->get_processed_providers_count() ) {
-			$this->state = self::SUCCESS;
+
+			// Only consider the generation a success if at least one provider was successful
+			if ( $this->get_providers_success_count() > 0 ) {
+				$this->state = self::SUCCESS;
+			} else {
+				$this->state = self::FAIL;
+			}
```

Comment:
Is this also run on non-cloud Critical CSS?

Reply:
Not sure I understand what you are asking. This is only referenced by cloud CSS at the moment.

Reply:
Critical_CSS state exists in both cloud and local versions. Digging a bit into the history - looks like a change happened on April 1 that introduced a change that only applies to the cloud generation. So my question was - why are we making changes that are specific to cloud css in a class that isn't specific to cloud css, but it turns out we've broken SRP already again.

Critical CSS Status class should have no knowledge about what types of critical css exist, be it cloud or local. It has become entangled once again.

Reply:
We do want to know the difference between the states or at least clear the existing state when critical css and cloud css switches in between.

How do you suggest we handle this case?

Reply:
If cloud and local css don't have the same generation css generation stages, should they share the same state tracking class?

I haven't looked too closely at this at the moment, but it looks like this might need untangling before it becomes even more complex.

Reply:
I think we should update the local css state to utilize more of the state.

Reply:
I don't think this PR is the right place to fix this, but I agree that having two modes of working with the Critical CSS state class is code smell.

I have filed a ticket for revisiting this and cleaning it up, and put it in the Next column so that this PR can be unblocked as I don't believe this PR increases the amount of code smell. 

47-gh-Automattic/boost-cloud